File: Demos/Todos/ObjectiveC/Todos/Serializer.m
//
//  Serializer.m
//  Toodoos
//
//  Created by Sasa Prodribaba on 13/12/2018.
//  Copyright © 2018 Sasa Prodribaba. All rights reserved.
//

#import "Serializer.h"
#import "Todo.h"

@implementation Serializer

- (NSArray *)serializeJSONObject:(NSData *)jsonData {
    
    NSError *jsonError;
    NSArray *jsonObjects = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:&jsonError];
    
    NSMutableArray *todos = [NSMutableArray new];
    
    if(!jsonError) {
        for (NSDictionary *dict in jsonObjects) {
            Todo *todo = [[Todo alloc] initWithDictionary:dict];
            [todos addObject:todo];
        }
    }
    
    return todos;
}

@end

File: Demos/Todos/Swift/Todos/Serializer.swift
//
//  Serializer.swift
//  Todos
//
//  Created by Sasa Prodribaba on 25/12/2018.
//  Copyright © 2018 Sasa Prodribaba. All rights reserved.
//

import Foundation

class Serializer {
    
    func serializeJSONObject(jsonData: Data) -> [Todo] {
        var todos = [Todo]()
        do {
            let jsonObjects = try JSONSerialization.jsonObject(with: jsonData, options: []) as? [[String: Any]]
            
            if let jsonObjects = jsonObjects {
                for dict in jsonObjects {
                    let todo = Todo(dict: dict)
                    todos.append(todo)
                }
            }
        } catch {
            print("JSON Error: \(error)")
        }
        return todos
    }
}
